modules/docs/docgen/spec/docParser.spec.ts

```diff
@@ -0,0 +1,1419 @@
+      export enum MyEnum {
+        Foo = 'foo',
+        Bar = 'bar',
+      }
+    `);
+    const docs = parse(program, 'test.ts');
+    expect(docs).toHaveProperty('0.name', 'MyEnum');
+    expect(docs).toHaveProperty('0.type.kind', 'object');
+    expect(docs).toHaveProperty('0.type.properties.0.kind', 'property');
+    expect(docs).toHaveProperty('0.type.properties.0.name', 'Foo');
+    expect(docs).toHaveProperty('0.type.properties.0.type.kind', 'string');
+    expect(docs).toHaveProperty('0.type.properties.0.type.value', 'foo');
+    expect(docs).toHaveProperty('0.type.properties.1.name', 'Bar');
+    expect(docs).toHaveProperty('0.type.properties.1.type.kind', 'string');
+    expect(docs).toHaveProperty('0.type.properties.1.type.value', 'bar');
+  });
+
+  it('should use the type checker evaluated value of local types', () => {
+    const program = createProgramFromSource(`
+      type ValueOf<T> = T[keyof T];
+      type Foo = {
+        foo: 'foo',
+        bar: 'bar',
+      }
+
+      export type Bar = ValueOf<Foo>
+    `);
+    const docs = parse(program, 'test.ts');
+
+    expect(docs).toHaveProperty('0.name', 'Bar');
+    expect(docs).toHaveProperty('0.type.kind', 'type');
+    expect(docs).toHaveProperty('0.type.value.kind', 'union');
+    expect(docs).toHaveProperty('0.type.value.value.0.kind', 'string');
+    expect(docs).toHaveProperty('0.type.value.value.0.value', 'foo');
+    expect(docs).toHaveProperty('0.type.value.value.1.kind', 'string');
+    expect(docs).toHaveProperty('0.type.value.value.1.value', 'bar');
+  });
+
+  it('should handle large unions', () => {
+    const program = createProgramFromSource(`
+      type Letters = 'a' | 'b' | 'c' | 'd' | 'e' | 'f' | 'g' | 'h' | 'i' | 'j' | 'k' | 'l' |  'm' | 'n' | 'o' | 'p' | 'q' | 'r' | 's' | 't' | 'u' | 'v' | 'w' | 'x' | 'y' | 'z'
+      type Numbers = '1' | '2'
+      type A = \`\${Letters}\${Numbers}\`
+
+      export type B = A
+    `);
+    const docs = parse(program, 'test.ts');
+
+    expect(docs).toHaveProperty('0.name', 'B');
+    expect(docs).toHaveProperty('0.type.kind', 'type');
+    expect(docs).toHaveProperty('0.type.value.kind', 'union');
+    expect(docs).toHaveProperty('0.type.value.value.length', 52);
+  });
+
+  it('should handle large unions', () => {
+    const program = createProgramFromSource(`
+      type Letters = 'a' | 'b' | 'c' | 'd' | 'e' | 'f' | 'g' | 'h' | 'i' | 'j' | 'k' | 'l' |  'm' | 'n' | 'o' | 'p' | 'q' | 'r' | 's' | 't' | 'u' | 'v' | 'w' | 'x' | 'y' | 'z'
+      type Numbers = '1' | '2'
+      type A = \`\${Letters}\${Numbers}\`
+
+      export type B = A
+    `);
+    const docs = parse(program, 'test.ts');
+
+    expect(docs).toHaveProperty('0.name', 'B');
+    expect(docs).toHaveProperty('0.type.kind', 'type');
+    expect(docs).toHaveProperty('0.type.value.kind', 'union');
+    expect(docs).toHaveProperty('0.type.value.value.length', 52);
+  });
+
+  it('should handle keyof from an enum', () => {
```

Comment:
This looks like a complete duplicate of the previous spec.

```suggestion
  it('should handle keyof from an enum', () => {
```

Reply:
Yep